### PR TITLE
Calibrate takeup rates

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Fixes #495 

- [x] Child Benefit
- [x] Child Tax Credit
- [x] Working Tax Credit
- [x] Universal Credit
- [x] Housing Benefit
- [x] Income Support
- [x] Income-based JSA
- [x] Pension Credit

### CODE REVIEW

File: openfisca_uk/parameters/benefit/JSA/income/takeup.yaml [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+description: Share of eligible Income-based Jobseeker's Allowance recipients that
+  participate
 metadata:
   label: Income-based Jobseeker's Allowance take-up rate
   name: JSA_income_takeup
-  unit: /1
   reference:
-    name: "Income-related benefits: estimates of take-up: financial year 2015/16 | Jobseeker's Allowance: tables | JSA1"
     href: https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/645577/income-related-benefits-estimates-of-take-up-2015-16.pdf
+    name: 'Income-related benefits: estimates of take-up: financial year 2015/16 |
```
Comment (MaxGhenis) [inline openfisca_uk/parameters/benefit/JSA/income/takeup.yaml]: would double-quotes be safer given the apostrophe?

File: openfisca_uk/tools/takeup.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+from openfisca_uk.system import CountryTaxBenefitSystem
+from openfisca_uk.tools.simulation import Microsimulation
+import logging
+import yaml
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Program:
+    """A government program with receipients who must choose to apply."""
```
Comment (MaxGhenis) [inline openfisca_uk/tools/takeup.py]:
```suggestion
    """A government program with recipients who must choose to apply."""
```

File: openfisca_uk/tools/takeup.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+    def __init__(
```
Comment (MaxGhenis) [inline openfisca_uk/tools/takeup.py:18]: docstring

File: openfisca_uk/tools/takeup.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            self.microsimulation.simulation.tax_benefit_system.variables[
+                self.variable
+            ].label
+        )
+        self.takeup = self._get_param(self.takeup_name)
+        self.caseload = self._get_param(self.caseload_name)
+        self.expenditure = self._get_param(self.expenditure_name)
+
+    def takeup_rate_error(self, takeup_rate: float, year: int) -> float:
+        if takeup_rate < 0 or takeup_rate > 1:
```
Comment (MaxGhenis) [inline openfisca_uk/tools/takeup.py:52]: Is this total takeup or takeup among households that are eligible but don't report claiming it? Total takeup can exceed 1.

File: openfisca_uk/tools/takeup.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    / "benefit"
+    / "tax_credits"
+    / "working_tax_credit"
+    / "takeup.yaml",
+)
+
+hb = Program(
+    "housing_benefit",
+    parameters.benefit.housing_benefit.statistics.caseload,
+    parameters.benefit.housing_benefit.statistics.expenditure,
```
Comment (MaxGhenis) [inline openfisca_uk/tools/takeup.py]: I'm noticing a pattern with these arguments :) Maybe Program could take the root parameter instead and always assume the `.statistics.caseload`, etc.?

File: openfisca_uk/tools/takeup.py (hunk 4 above)
Comment (nikhilwoodruff) [reply openfisca_uk/tools/takeup.py:52]: Just those that don't report it - will update the descriptions

File: openfisca_uk/tools/takeup.py (hunk 5 above)
Comment (nikhilwoodruff) [reply openfisca_uk/tools/takeup.py]: Yeah I think I see :D - consolidated

File: openfisca_uk/parameters/benefit/JSA/income/takeup.yaml (hunk 1 above)
Comment (nikhilwoodruff) [reply openfisca_uk/parameters/benefit/JSA/income/takeup.yaml]: Removed entirely given parameters are no longer directly comparable

File: openfisca_uk/tools/takeup.py (hunk 4 above)
Comment (nikhilwoodruff) [reply openfisca_uk/tools/takeup.py:52]: But although total takeup can exceed 1, we can't model it, as it just informs a probability

**Comment by nikhilwoodruff** [comment]
@MaxGhenis I'm noticing that some legacy benefits are prevented from matching closely (WTC, CTC in recent years) because they're simulated after UC, and I guess the recipients left over after UC selection aren't enough for the WTC/CTC optimiser to work with.

**Comment by nikhilwoodruff** [comment]
Closing in favour of #505 - but filing #506.

**Pull request closed without merging**